File: examples/src/main/java/com/superior/datatunnel/examples/jdbc/DataTunnelOracle2LogDemo.java
package com.superior.datatunnel.examples.jdbc;

import com.superior.datatunnel.core.DataTunnelExtensions;
import org.apache.spark.sql.SparkSession;

import java.sql.SQLException;

public class DataTunnelOracle2LogDemo {

    public static void main(String[] args) throws SQLException {
        SparkSession spark = SparkSession
                .builder()
                //.enableHiveSupport()
                .master("local")
                .appName("Datatunnel spark example")
                .config("spark.sql.extensions", DataTunnelExtensions.class.getName())
                .getOrCreate();

        String sql = "datatunnel SOURCE('oracle') OPTIONS(\n" +
                "    username='flinkuser',\n" +
                "    password='flinkpw',\n" +
                "    host='172.18.1.56',\n" +
                "    port=1521,\n" +
                "    serviceName='XE',\n" +
                "    databaseName='FLINKUSER'," +
                "    tableName='ORDERS', columns=['ORDER_ID', 'CUSTOMER_NAME', 'PRICE', 'PRODUCT_ID', 'ORDER_STATUS'])\n" +
                "    SINK('log') OPTIONS(numRows = 10)";

        spark.sql(sql);
    }
}

File: examples/src/main/kotlin/com/superior/datatunnel/examples/jdbc/DataTunnelOracle2LogDemo.kt
package com.superior.datatunnel.examples.jdbc

import com.superior.datatunnel.core.DataTunnelExtensions
import org.apache.spark.sql.SparkSession

object DataTunnelOracle2LogDemo {

    @JvmStatic
    fun main(args: Array<String>) {
        System.setProperty("HADOOP_USER_NAME", "hdfs");

        val spark = SparkSession
            .builder()
            .master("local")
            .enableHiveSupport()
            .appName("Datatunnel spark example")
            .config("spark.sql.extensions", DataTunnelExtensions::class.java.name)
            .getOrCreate()

        val sql = """
            DATATUNNEL SOURCE('oracle') OPTIONS (
              username = 'system',
              password = 'system',
              jdbcUrl = 'jdbc:oracle:thin:@//172.18.1.51:1523/XE',
              databaseName = 'SYSTEM',
              schemaName = 'FLINKUSER',
              tableName = 'Oracle_paimon_spark_oracle',
              columns = ['ID', 'AGE', 'NAME', 'IS_ACTIVE'],
              sourceTempView = 'preview_view'
            ) 
            SINK("log") 
        """.trimIndent()

        spark.sql(sql)
    }
}